benchmark/conftest.py

```diff
@@ -389,3 +390,12 @@ def pytest_generate_tests(metafunc):
             "test_data_text_extraction",
             data_file["text_extraction"],
         )
+
+    os.chdir("benchmark")
```

Comment:
The key is in the form `./data/filename.yaml`. The program root is the project root. Thus the `benchmark` is missing in the path to access files. This was just a quick workaround to avoid changing too many things. Maybe adapting the string would be nicer?
A few lines later the working dir is changed to the default again.

Reply:
hm, why do we even use a directory structure in the dictionary keys in the first place? can't we switch to just using the name, and then deal with paths in the function that require them?

Reply:
that includes the suffix, not sure this adds any value in the dict keys

Reply:
btw I was referring to the entire procedure, why do we write these yaml files here? at least should be documented in the docstring what this achieves

Reply:
Maybe we could just keep everything in one yaml, i.e., make the schema definition(s) part of the test dataset yaml? Didn't think this through, but seems at least possible.

Reply:
Yes, this is definitely possible.
But in the case with more test cases and more schema definitions the `benchmark_data.yaml` can easily grow up to > 1000 lines with the risk of getting difficult to read.
Thus, for now, I would keep it separated.

Reply:
Good point, but in this case I would probably prefer splitting the YAMLs by topic, context, or module. Could also have one for each of the different schema definitions. We are already reading them in automatically, so that should already work as is, right?

Reply:
Yes, splitting by topic makes more sense to handle this.